src/lib.rs

```diff
@@ -139,9 +192,6 @@ enum DecoderState {
 
     /// 255 bytes, will be a header next
     GrabChain(u8),
-
-    /// Prevent re-using the state machine
-    ErrOrComplete,
```

Comment:
Oh, one (potentially?) breaking change, before I wouldn't let you re-use the buffer, but now the state machine will go back to Idle once a message is completed. Let me know if you'd like to make this backwards compatible, or just issue a 0.2.0 release to make the breaking change okay.

Otherwise I would probably need a `completed` flag in `CobsDecoder` to prevent reuse (probably a good thing, since that is based on a buffer).